xbmc/addons/Repository.h

```diff
@@ -59,25 +57,33 @@ namespace ADDON
     static bool Parse(const DirInfo& dir, VECADDONS& addons);
     static std::string FetchChecksum(const std::string& url);
 
-    virtual void OnPostInstall(bool update, bool modal);
-    virtual void OnPostUnInstall();
-
   private:
     CRepository(const CRepository &rhs);
   };
 
+  typedef std::shared_ptr<CRepository> RepositoryPtr;
+
+
   class CRepositoryUpdateJob : public CProgressJob
   {
   public:
-    CRepositoryUpdateJob(const VECADDONS& repos);
+    CRepositoryUpdateJob(const RepositoryPtr& repo);
     virtual ~CRepositoryUpdateJob() {}
-
-    virtual const char *GetType() const { return "repoupdate"; };
     virtual bool DoWork();
+    const RepositoryPtr& GetAddon() const { return m_repo; };
+
   private:
-    bool GrabAddons(const RepositoryPtr& repo, VECADDONS& addons);
+    enum class FetchStatus
+    {
+      OK,
+      NOT_MODIFIED,
+      ERROR
```

Comment:
The problem most likely is that `ERROR` is already a global define or typedef provided by some windows API or third party library.